test/test_torch.py

```diff
@@ -1695,8 +1695,7 @@ def test_add(self):
         m1 = torch.tensor([True, False, False, True, False, False], dtype=torch.bool)
         m2 = torch.tensor([True, True, False, False, False, True], dtype=torch.bool)
         self.assertEqual(m1 + m2, torch.tensor([True, True, False, True, False, True], dtype=torch.bool))
-
-        # [res] torch.add([res,] tensor1, value, tensor2)
+        self.assertEqual(m1.to('cuda') + m2.to('cuda'), torch.tensor([True, True, False, True, False, True], device='cuda', dtype=torch.bool))
```

Comment:
This isn't going to work on non-CUDA builds. Either guard it with a torch.cuda.is_available() or take in a cast fn like some of the other tests do

Reply:
Or maybe `for device in torch.testing.get_all_device_types():` is the preferred way now? I'm not sure. @gchanan would know.

Reply:
Yeah, i was about to suggest. I introduced orch.testing.get_all_device_types() as its a bit confusing to test same thing from 2 different test files. Also, test_cuda is really confusing and im planning to take a deep dive there and simplify the whole testing thing